lib/hyperswitch.js

```diff
@@ -54,7 +54,12 @@ function HyperSwitch(options, req, parOptions) {
         this._rootReq.headers = this._rootReq.headers || {};
         this._requestFilters = par._requestFilters;
         this._subRequestFilters = par._subRequestFilters;
-        this.ctx = par.ctx || {};
+        this.ctx = par.ctx || {
+            headers: {
+                'user-agent': req.headers['user-agent'],
+                'x-client-ip': req.headers['x-client-ip'],
```

Comment:
The issue with XFF is that it's basically impossible to parse correctly without a lot of information about the cache layers. Backend services typically only care about x-client-ip for rate limiting / blocking purposes, for which x-client-ip is the better choice.